### PR TITLE
Add dashboard

### PR DESCRIPTION
**Pull request by klippx**
Add dashboard showing failure count. This depends on https://github.com/klarna/phobos_db_checkpoint/pull/27

- [x] When this is merged we should update gemspec to `spec.add_dependency 'phobos_db_checkpoint', '~> 2.4'`

screen shot 2017-03-13 at 14 49 44

### CODE REVIEW

File: frontend/src/actions/dashboard/failure-count/index.spec.js [hunk 1]
```diff
@@ -0,0 +1,70 @@
...
+  describe('#fetchFailureCount', () => {
+    describe('when it succeeds', () => {
+      let count, store
+      beforeEach(() => {
+        count = 7
+        store = mockStore({})
+        Mappersmith.Env.Fixture
+          .define('get')
+          .matching({ url: '/api/v1/failures/count' })
+          .response(count)
```
Comment (ticolucci) [inline frontend/src/actions/dashboard/failure-count/index.spec.js]: The new api replies an object instead:
```javascript
{ "count": 7 }
```

File: frontend/src/actions/dashboard/failure-count/index.spec.js [hunk 2]
```diff
@@ -0,0 +1,70 @@
...
+      })
+
+      it('creates REQUEST and RECEIVE actions', (done) => {
+        store.dispatch(fetchFailureCount()).then(() => {
+          const actions = store.getActions()
+          expect(actions[0]).toEqual({ type: REQUEST_FAILURE_COUNT })
+          expect(actions[1]).toEqual({ type: RECEIVE_FAILURE_COUNT, data: count })
```
Comment (ticolucci) [inline frontend/src/actions/dashboard/failure-count/index.spec.js]: Even though the response has an object. I don't mind at all parsing it to the number in the action (not the reducer). 
So this spec, in my opinion, should stay like this still ;-)

File: frontend/src/views/dashboard/index.js [hunk 3]
```diff
@@ -0,0 +1,55 @@
...
+      fetchFailureCount: PropTypes.func.isRequired,
+
+      xhrStatus: PropTypes.shape({
+        isFetchingFailureCount: PropTypes.bool
+      })
+    }
+  }
+
+  componentDidMount () {
+    !this.props.failureCount &&
```
Comment (ticolucci) [inline frontend/src/views/dashboard/index.js]: I would prefer for the dashboard to be long-pooling for the failure count. 
This way we don't have to refresh the page by browser extensions, etc in the monitors.
(and this is super easy to achieve w/ setInterval =P)

File: frontend/src/reducers/dashboard/index.spec.js [hunk 4]
```diff
@@ -0,0 +1,16 @@
...
+  RECEIVE_FAILURE_COUNT
+} from 'actions'
+
+import reducer from 'reducers/dashboard'
+
+describe('reducers/dashboard', () => {
+  describe('for RECEIVE_FAILURE_COUNT', () => {
+    it('sets overviewVisible for the specific event to true', () => {
+      const currentState = {}
+      const action = { type: RECEIVE_FAILURE_COUNT, data: { count: 8 } }
```
Comment (ticolucci) [inline frontend/src/reducers/dashboard/index.spec.js]: just noticed that you parsed the count in the reducer... >_<
choose one an go for it. I don't have a strong opinion. Just need to update the action spec then.

File: frontend/src/views/dashboard/index.js (hunk 3 above)
Comment (klippx) [inline frontend/src/views/dashboard/index.js]: This is also where the idea of websocket gateway for mappersmith comes in :-)

Comment (ticolucci) [inline frontend/src/views/dashboard/index.js]: I agree. Websockets would be perfect for this use-case, but while we don't have them... ;-P

File: frontend/src/reducers/dashboard/index.spec.js (hunk 4 above)
Comment (klippx) [inline frontend/src/reducers/dashboard/index.spec.js]: Done

File: frontend/src/views/dashboard/index.js (hunk 3 above)
Comment (klippx) [inline frontend/src/views/dashboard/index.js]: Done!

**Pull request merged**